docs/usage/shoot_updates.md

```diff
@@ -85,6 +88,9 @@ The complete list of fields that trigger a rolling update:
 * `.spec.provider.workers[].providerConfig`
```

Comment:
How about to mention `providerConfig` is not included in v2?

For the sake of clarity, it might be worth creating two separate lists, one for v1 and one fore v2?

Reply:
I've extracted the `NewWorkerPoolHash` specific fields into a separate block.

If `NewWorkerPoolHash` is enabled and a shoot has been reconciled once, then changes to fields included only in v2 will also trigger a node roll. (The v1 and v2 hashes are by then included in the "worker-pools-operatingsystemconfig-hashes" secret, which allows detecting the v2 hash change, even though the worker pool name still uses v1). From the user perspective only the `providerConfig` will behave slightly different, as it is no longer included in v2 hashes but requires extensions to explicitly decide which fields of the providerConfig to include. So, from the perspective of a user it's mostly relevant whether the feature gate is enabled; whereas the hash version used by a worker pool is not really relevant (only for the providerConfig field).